argusclient/client.py

```diff
@@ -139,19 +142,27 @@ def __init__(self, argus):
 
 
 class BaseModelServiceClient(object):
-    def __init__(self, argus, get_all_path=None, get_all_params=None):
+    def __init__(self, argus, get_all_path=None,
+                                get_all_params=None,
+                                get_all_request_type=None,
+                                get_all_body=None):
         self.argus = argus
         self._retrieved_all = False
         self._coll = {}
         self.get_all_path = get_all_path
         self.get_all_params = get_all_params
+        self.get_all_request_type = get_all_request_type or "get"
+        self.get_all_body = get_all_body
 
     def _init_all(self, coll=None):
         if not self.get_all_path:
             raise TypeError("Unsupported operation on: %s" % type(self))
         if not self._retrieved_all:
             self._coll = dict((obj.argus_id, self._fill(obj))
-                                for obj in coll or self.argus._request("get", self.get_all_path, params=self.get_all_params))
+                              for obj in (coll or self.argus._request(self.get_all_request_type,
+                                                                       self.get_all_path,
+                                                                       params=self.get_all_params,
+                                                                       dataObj = self.get_all_body)))
```

Comment:
Continuing with the previous suggestion,
```suggestion
                              for obj in (coll or self.argus._request(self.get_all_req_opts.get(REQ_METHOD, "get"),
                                                                       self.get_all_req_opts.get(REQ_PATH),
                                                                       params=self.get_all_req_opts.get(REQ_PARAMS),
                                                                       dataObj = self.get_all_req_opts.get(REQ_BODY))))
```

The check on Line 158 will need to change as well.

For some reason, the above is getting formatted badly, so inserting it again without a suggestion:

```
                              for obj in (coll or self.argus._request(self.get_all_req_opts.get(REQ_METHOD, "get"),
                                                                       self.get_all_req_opts.get(REQ_PATH),
                                                                       params=self.get_all_req_opts.get(REQ_PARAMS),
                                                                       dataObj = self.get_all_req_opts.get(REQ_BODY))))
```